monitoring/asset_collector.go

```diff
@@ -0,0 +1,125 @@
+package monitoring
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	numAssetsMintedMetric = "num_assets_minted"
+
+	numTotalGroupsMetric = "num_total_groups"
+
+	numTotalSyncsMetric = "num_total_syncs"
+
+	numTotalProofsMetric = "num_total_proofs"
+)
+
+// universeStatsCollector is a Prometheus collector that exports the stats of
+// the universe.
+type universeStatsCollector struct {
+	collectMx sync.Mutex
+
+	cfg      *PrometheusConfig
+	registry *prometheus.Registry
+
+	gauges map[string]prometheus.Gauge
+}
+
+func newUniverseStatsCollector(cfg *PrometheusConfig,
+	registry *prometheus.Registry) *universeStatsCollector {
+
+	if cfg == nil {
+		log.Error("cfg is nil")
```

Comment:
Instead of logging, just return an `fmt.Errorf` which will then prevent the daemon from starting (instead of logging something we'd skip over). (For all other checks as well).

Reply:
that's what I'm currently doing by returning `nil` (caller checks if returned struct is `nil`, if yes it errors out)

Reply:
can switch to `error` ofcs if it's more readable